src/android/MediaCapture.java

```diff
@@ -204,48 +71,55 @@ public void run() {
         }
     }
 
-    @Override
-    public void onPause(boolean multitasking) {
-        if (previewing) {
-            this.appPausedWithActivePreview = true;
-            this.pausePreview(null);
+    private void openNewActivity(Context context) {
+
+        if (!hasPermission()) {
+            requestPermission(33);
+            return;
+        } else {
+            Intent intent = new Intent(context, MainActivity.class);
+            intent.putExtra("VIDEO_MAX_LENGTH", lengthInSeconds);
+            intent.putExtra("NEXT_VIDEO_URL", createCaptureFile().getAbsolutePath());
+            cordova.startActivityForResult((CordovaPlugin) this ,intent,VIDEO_URL);
         }
     }
 
-    @Override
-    public void onResume(boolean multitasking) {
-        if (this.appPausedWithActivePreview) {
-            this.appPausedWithActivePreview = false;
-            this.resumePreview(null);
-        }
+    private File createCaptureFile() {
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        return new File(getTempDirectoryPath(),
+                "vid_" + timeStamp + ".mp4");
     }
 
-    private boolean hasFlash() {
-        if (flashAvailable == null) {
-            flashAvailable = false;
-            final PackageManager packageManager = this.cordova.getActivity().getPackageManager();
-            for (final FeatureInfo feature : packageManager.getSystemAvailableFeatures()) {
-                if (PackageManager.FEATURE_CAMERA_FLASH.equalsIgnoreCase(feature.name)) {
-                    flashAvailable = true;
-                    break;
-                }
-            }
+    private String getTempDirectoryPath() {
+        File cache = null;
+
+        // SD Card Mounted
+        if (Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)) {
+            cache = cordova.getActivity().getExternalCacheDir();
         }
-        return flashAvailable;
+        // Use internal storage
+        else {
+            cache = cordova.getActivity().getCacheDir();
+        }
+
+        // Create the cache directory if it doesn't exist
+        cache.mkdirs();
+        return cache.getAbsolutePath();
```

Comment:
are those files clear in situations of:

- [ ] video file successfully recorded. After it is moved to app directory, is the original file deleted?
- [ ] video file recording started but not finished. application crashed/killed by user or system. is the original file deleted, after relunch of app, or this java component?

Reply:
fixed